eclide/ChildBuilderFrame.cpp

```diff
@@ -404,7 +408,27 @@ class CBuilderFrame :
             IAttributeVector attrs;
             Dali::CEclExceptionVector errors;
             IAttributeBookkeep attrProcessed;
-            attr->PreProcess(PREPROCESS_SUBMIT, _ecl, attrs, attrProcessed, errors);
+            PREPROCESS_TYPE processType = PREPROCESS_UNKNOWN;
+            switch (action) {
+            case Dali::WUActionGenerate:
+                if (!isSaved)
+                    m_dlgview.DoGenerate();
+                processType = PREPROCESS_GENERATE;
+                break;
+            case Dali::WUActionCustom1:
```

Comment:
@rpastrana duely noted. This is semantically true but it is part of the enum type WUAction in Dali.h and although it may not be the best place, the separation for semantics sake is not worth the effort at this time.